### PR TITLE
Validate that requests are coming from Twilio (CU-dgmfbv)

### PR DESCRIPTION
**Pull request by risatronic**
- Implement isValidTwilioRequest function in twilio.js to be used in handleTwilioRequest method of braveAlerter
- Update existing tests to mock isValidTwilioRequest returning true/false as needed
- Add unit tests for case where isValidTwilioRequest returns false (response status should be 401)

### CODE REVIEW

**Comment by trisapeace** [comment]
Task linked: [CU-dgmfbv Validate that requests from Twilio really come from Twilio](https://app.clickup.com/t/dgmfbv)

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- Add Twilio validation to make sure that post requests are coming from Twilio and relevant tests(CU-dgmfbv).
```
Comment (trisapeace) [inline CHANGELOG.md:9]: Looks like you're missing a space before the brackets around the ClickUp number.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
 
+### Updated
+- Updated existing tests to work with implementation of Twilio validation (CU-dgmfbv).
```
Comment (trisapeace) [inline CHANGELOG.md:12]: No need to mention updating the tests in the changelog. In theory, every change will included updating tests :smile:

File: lib/twilio.js [hunk 3]
```diff
@@ -34,6 +38,7 @@ async function sendTwilioResponse(response, message) {
 }
 
 module.exports = {
+    isValidTwilioRequest,
     sendTwilioMessage,
-    sendTwilioResponse,
```
Comment (trisapeace) [inline lib/twilio.js:38]: It's actually good to keep these trailing commas in lists like this that change frequently. It makes the diff easier to understand when looking at pull requests or looking back on commit changes. Here's a little more info: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas

File: package-lock.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "brave-alert-lib",
-  "version": "1.0.0",
```
Comment (trisapeace) [inline package-lock.json:3]: [This comment is not directly related to your pull request.]

Hmmm, this is interesting to note. I didn't realize that it would show up here too. This version number change is coming from `package.json` in the `main` branch. We'll need to remember for the next release that we need to update it here too.

**Review by trisapeace — COMMENTED** [review]
Looks good. Thanks for sticking with this. I hope that next time will be easier to wade through the tests.

File: package-lock.json (hunk 4 above)
Comment (risatronic) [reply package-lock.json:3]: It says 2 on my local repo in there, that's strange 🤔

File: CHANGELOG.md (hunk 1 above)
Comment (risatronic) [reply CHANGELOG.md]: Fixed & removed the bit about tests 😅

File: lib/twilio.js (hunk 3 above)
Comment (risatronic) [reply lib/twilio.js:38]: Fixed :)

File: package-lock.json (hunk 4 above)
Comment (trisapeace) [reply package-lock.json:3]: 2 is good :+1:  That's what it should say. I just missed it when I was updating the version last time I merged into `main`.

**Review by trisapeace — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,7 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
-
+### Added
```
Comment (trisapeace) [inline CHANGELOG.md:8]: I just remembered, in the Changelog specs (https://keepachangelog.com/en/1.0.0/) it does say that there is a heading for "Security". That would probably be better to use here instead of putting it under "Added".

Comment (risatronic) [reply CHANGELOG.md]: Done! :)

**Review by schwarrrtz — APPROVED** [review]
This looks good. I haven't done a full end-to-end test but I did run the automated tests - once @risatronic reminded me about the Twilio test credentials, I was able to get everything to pass.

**Comment by schwarrrtz** [comment]
also @trisapeace @risatronic I pushed a commit that adds a comment to `.env.example` regarding the test credentials.

**Comment by trisapeace** [comment]
Cool. @risatronic do you have 1/2 hour tomorrow that we could go through the deployment of the library and up-ing its version together?

**Comment by risatronic** [comment]
> Cool. @risatronic do you have 1/2 hour tomorrow that we could go through the "deployment" of the library and up-ing its version together?

I do yes! Should it stay unmerged for now?

**Comment by trisapeace** [comment]
You can go ahead and merge. I suggest you do the "Squash and Merge" option so that it's just one commit.
